lib/puppetserver/ca/action/prune.rb

```diff
@@ -106,6 +120,50 @@ def update_pruned_CRL(crl, pkey)
           crl.sign(pkey, OpenSSL::Digest::SHA256.new)
         end
 
+        def prune_using_serial(crl, key, serialnumbers)
+          removed_serials = []
+          revoked_list = crl.revoked
+          @logger.debug("Removing entries in CRL for issuer " \
+            "#{crl.issuer.to_s(OpenSSL::X509::Name::RFC2253)}") if @logger.debug?
+          serialnumbers.each do |serial|
+            revoked_list.delete_if do |revoked|
+              if revoked.serial == OpenSSL::BN.new(serial.hex)
```

Comment:
We should probably validate that the serial numbers are actual valid hex representations, since something like `'absolutely'.hex` gives you `171` and `'one'.hex` gives you `0`.

Reply:
Probably use `.match(/^(?:0x)?[A-Fa-f0-9]+$/)`

Reply:
Updated